lib/datadog/grok/src/grok_filter.rs

```diff
@@ -0,0 +1,236 @@
+    Json,
+    Rubyhash,
+    Querystring,
+    Lowercase,
+    Uppercase,
+    Decodeuricomponent,
+}
+
+impl TryFrom<&Function> for GrokFilter {
+    type Error = GrokStaticError;
+
+    fn try_from(f: &Function) -> Result<Self, Self::Error> {
+        match f.name.as_str() {
+            "boolean" => {
+                if f.args.is_some() && !f.args.as_ref().unwrap().is_empty() {
+                    if let FunctionArgument::Arg(Value::Bytes(ref bytes)) =
+                        f.args.as_ref().unwrap()[0]
+                    {
+                        let pattern = String::from_utf8_lossy(bytes);
+                        Ok(GrokFilter::Boolean(Some(
+                            Regex::new(pattern.deref()).map_err(|error| {
+                                error!(message = "Error compiling regex", path = %pattern, %error);
+                                GrokStaticError::InvalidFunctionArguments(f.name.clone())
+                            })?,
+                        )))
+                    } else {
+                        Err(GrokStaticError::InvalidFunctionArguments(f.name.clone()))
+                    }
+                } else {
+                    Ok(GrokFilter::Boolean(None))
+                }
+            }
+            "nullIf" => {
+                if f.args.is_some() && !f.args.as_ref().unwrap().is_empty() {
+                    if let FunctionArgument::Arg(ref null_value) = f.args.as_ref().unwrap()[0] {
+                        return Ok(GrokFilter::NullIf(null_value.to_string_lossy()));
+                    }
+                }
+                Err(GrokStaticError::InvalidFunctionArguments(f.name.clone()))
+            }
+            "scale" => {
+                if f.args.is_some() && !f.args.as_ref().unwrap().is_empty() {
+                    let scale_factor = match f.args.as_ref().unwrap()[0] {
+                        FunctionArgument::Arg(Value::Integer(scale_factor)) => scale_factor as f64,
+                        FunctionArgument::Arg(Value::Float(scale_factor)) => scale_factor,
+                        _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
+                    };
+                    return Ok(GrokFilter::Scale(scale_factor));
+                }
+                Err(GrokStaticError::InvalidFunctionArguments(f.name.clone()))
+            }
+            "integer" => Ok(GrokFilter::Integer),
+            "integerExt" => Ok(GrokFilter::IntegerExt),
+            "number" => Ok(GrokFilter::Number),
+            "numberExt" => Ok(GrokFilter::NumberExt),
+            "json" => Ok(GrokFilter::Json),
+            "rubyhash" => Ok(GrokFilter::Rubyhash),
+            "querystring" => Ok(GrokFilter::Querystring),
+            "lowercase" => Ok(GrokFilter::Lowercase),
+            "uppercase" => Ok(GrokFilter::Uppercase),
+            "decodeuricomponent" => Ok(GrokFilter::Decodeuricomponent),
+            _ => Err(GrokStaticError::UnknownFilter(f.name.clone())),
+        }
+    }
```

Comment:
```suggestion
    fn try_from(f: &Function) -> Result<Self, Self::Error> {
        match f.name.as_str() {
            "boolean" => match f.args.as_ref() {
                Some(args) if !args.is_empty() => match args[0] {
                    FunctionArgument::Arg(Value::Bytes(ref bytes)) => {
                        let pattern = String::from_utf8_lossy(bytes);
                        Ok(GrokFilter::Boolean(Some(
                            Regex::new(pattern.deref()).map_err(|error| {
                                error!(message = "Error compiling regex", path = %pattern, %error);
                                GrokStaticError::InvalidFunctionArguments(f.name.clone())
                            })?,
                        )))
                    }
                    _ => Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
                },
                _ => Ok(GrokFilter::Boolean(None)),
            },
            "nullIf" => f
                .args
                .as_ref()
                .and_then(|args| {
                    if let FunctionArgument::Arg(ref null_value) = args[0] {
                        return Some(GrokFilter::NullIf(null_value.to_string_lossy()));
                    } else {
                        None
                    }
                })
                .ok_or_else(|| GrokStaticError::InvalidFunctionArguments(f.name.clone())),
            "scale" => match f.args.as_ref() {
                Some(args) if !args.is_empty() => {
                    let scale_factor = match args[0] {
                        FunctionArgument::Arg(Value::Integer(scale_factor)) => scale_factor as f64,
                        FunctionArgument::Arg(Value::Float(scale_factor)) => scale_factor,
                        _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
                    };
                    Ok(GrokFilter::Scale(scale_factor))
                }
                _ => Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
            },
            "integer" => Ok(GrokFilter::Integer),
            "integerExt" => Ok(GrokFilter::IntegerExt),
            "number" => Ok(GrokFilter::Number),
            "numberExt" => Ok(GrokFilter::NumberExt),
            "json" => Ok(GrokFilter::Json),
            "rubyhash" => Ok(GrokFilter::Rubyhash),
            "querystring" => Ok(GrokFilter::Querystring),
            "lowercase" => Ok(GrokFilter::Lowercase),
            "uppercase" => Ok(GrokFilter::Uppercase),
            "decodeuricomponent" => Ok(GrokFilter::Decodeuricomponent),
            _ => Err(GrokStaticError::UnknownFilter(f.name.clone())),
        }
    }
```

Slightly rewritten to lean more on the type system for exhaustive checking. I find the `.is_some() && .unwrap()` pattern a little disjointed.